### PR TITLE
Sapling Turnstile automatic migration

### PR DESCRIPTION
**Pull request by adityapk00**

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -16,6 +18,8 @@ int main(int argc, char *argv[])
         qApp->setFont(QFont("Ubuntu", 11, QFont::Normal, false));
     #endif
 
+	std::srand(std::time(nullptr));
```
Comment (daira) [inline src/main.cpp]: This is not cryptographically secure.

File: src/turnstile.cpp [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+		// Also account for the fees needed to send all these transactions
+		auto actual = amount - (Utils::getMinerFee() * (amounts.size() + 1));
+
+		amounts.push_back(actual);
+		return;
+	}
+
+	// Get a random amount off the amount (between half and full) and call recursively.
+	// Multiply by hundered, because we'll operate on 0.01 ZEC minimum. We'll divide by 100 later
+	double curAmount = std::rand() % (int)std::floor(amount * 100);
```
Comment (daira) [inline src/turnstile.cpp]: This doesn't appear to be doing what the comment says ("between half and full").

Comment (adityapk00) [reply src/turnstile.cpp]: Thanks for looking at this. I really appreciate it. 

Just note that the latest code lives on master branch (I've incorporated a bunch of feedback there, so please look at turnstile.cpp) here: https://github.com/adityapk00/zec-qt-wallet/blame/master/src/turnstile.cpp

File: src/main.cpp (hunk 1 above)
Comment (adityapk00) [reply src/main.cpp]: Fixed in b150f074f7ad2baad3179510b5f716cc1ad79bb6

File: src/turnstile.cpp (hunk 2 above)
Comment (adityapk00) [reply src/turnstile.cpp]: Fixed in b150f074f7ad2baad3179510b5f716cc1ad79bb6

**Pull request merged**